File: q1288/q1288.java
import java.util.Arrays;

class Solution {
    public int removeCoveredIntervals(int[][] intervals) {
        Arrays.sort(intervals, (i1, i2) -> {
            if (i1[0] == i2[0]) {
                return Integer.compare(i2[1], i1[1]);
            } else {
                return Integer.compare(i1[0], i2[0]);
            }
        });

        int res = 0;
        int prevEnd = -1;
        for (int[] interval : intervals) {
            int newEnd = interval[1];
            if (newEnd <= prevEnd) {
                res += 1;
            } else {
                prevEnd = newEnd;
            }
        }
        return intervals.length - res;
    }
}

File: q1288/q1288.py
class Solution:
    def removeCoveredIntervals(self, intervals: List[List[int]]) -> int:
        intervals.sort(key=lambda x: (x[0], -x[1]))
        res = 0
        prev_end = -1

        for interval in intervals:
            if interval[1] <= prev_end:
                res += 1
            else:
                prev_end = interval[1]

        return len(intervals) - res
